apollo-sample/src/main/java/com/example/apollographql/sample/MainActivity.java

```diff
@@ -23,39 +27,60 @@
     super.onCreate(savedInstanceState);
     setContentView(R.layout.activity_main);
     final TextView txtResponse = (TextView) findViewById(R.id.txt_response);
-    SampleApplication application = (SampleApplication) getApplication();
-    application.githuntApiService()
-        .githuntFeed(new OperationRequest<>(
-            new FeedQuery(FeedQuery.Variables.builder()
-                .limit(10)
-                .type(FeedType.HOT)
-                .build())))
+    final RecyclerView responses = (RecyclerView) findViewById(R.id.responses);
+
+    final SampleApplication application = (SampleApplication) getApplication();
+
+    final Subject<Integer> upvoteObserver = PublishSubject.create();
```

Comment:
It feels unnecessary for the activity to create the subject. Might make sense to have adapter create subject and the activity can do adapter.getUpdater.subscribe